tests/StatementTest.php

```diff
@@ -39,24 +39,21 @@ public function testInstantiation() {
 
     public function testFromJSONInvalidNull() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_NONE
+            'TinCan\JsonParseErrorException'
         );
         $obj = Statement::fromJSON(null);
     }
 
     public function testFromJSONInvalidEmptyString() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_NONE
+            'TinCan\JsonParseErrorException'
         );
         $obj = Statement::fromJSON('');
     }
 
     public function testFromJSONInvalidMalformed() {
         $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_SYNTAX
+            'TinCan\JsonParseErrorException'
```

Comment:
We would effectively be testing how json_encode() behaves under certain types of input which is already done for us [e.g](https://github.com/php/php-src/tree/master/ext/json/tests).

All we need to test is whether our exception returns a message that is formatted how we want that includes output from json_last_error(). That is done in JsonParseErrorExceptionTest::testThatWeGetAMeaningfulErrorMessage().